recipes/qwt/all/conanfile.py

```diff
@@ -0,0 +1,110 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+class QwtConan(ConanFile):
+    name = "qwt"
+    license = "Qwt License, Version 1.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://qwt.sourceforge.io/"
+    topics = ("conan", "archive", "compression")
+    description = (
+        "The Qwt library contains GUI Components and utility classes which are primarily useful for programs "
+        "with a technical background. Beside a framework for 2D plots it provides scales, sliders, dials, compasses, "
+        "thermometers, wheels and knobs to control or display values, arrays, or ranges of type double."
+    )
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "plot": [True, False],
+        "widgets": [True, False],
+        "svg": [True, False],
+        "opengl": [True, False],
+        "mathml": [True, False],
+        "designer": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "plot": True,
+        "widgets": True,
+        "opengl": True,
+        "designer": True,
+        "mathml": False,
+        "svg": False
+
+    }
+    generators = "qmake", "cmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("qt/5.15.1")
+
+    def configure(self):
+        if self.settings.os == "Windows" and self.settings.compiler != "Visual Studio":
+            raise ConanInvalidConfiguration("Only msvc compiler is supported on windows")
```

Comment:
qwt should support mingw: https://qwt.sourceforge.io/qwtinstall.html

Reply:
how to add support for mingw, is there any recipe that support it already ?

Reply:
What do you mean by "any recipe"? Most recipes in CCI support MinGW out of the box.

For qwt, mingw32-make should be called (instead of nmake) as explained in documentation.

Reply:
but should I add mingw as build requirements or it is supposed to be installed at the system level ?

Reply:
It(s not required and supposed to be installed at system level.